third_party/dependency_analyzer/src/test/io/bazel/rulesscala/dependencyanalyzer/ScalacDependencyTest.scala

```diff
@@ -0,0 +1,105 @@
+package third_party.dependency_analyzer.src.test.io.bazel.rulesscala.dependencyanalyzer
+
+import java.nio.file.Files
+import java.nio.file.Path
+import java.util.UUID
+import org.apache.commons.io.FileUtils
+import org.scalatest._
+import third_party.utils.src.test.io.bazel.rulesscala.utils.JavaCompileUtil
+import third_party.utils.src.test.io.bazel.rulesscala.utils.TestUtil
+
+/**
+ * Test that the scalac compiler behaves how we expect it to around
+ * dependencies. That is, for given scenarios, we want to make sure
+ * that scalac requires the given set of dependencies; no more and
+ * no less.
```

Comment:
WDYT about adding a sentence at the end saying something like:
```
To clarify- these tests do not reflect the end result of strict-deps/unused-deps as we're considering alternatives of how to mitigate scalac's limitations.
```
Suggesting this mainly since the purpose of this test wasn't 100% clear to me